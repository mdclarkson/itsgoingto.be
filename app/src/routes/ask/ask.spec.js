import React from 'react'
import { render } from '@testing-library/react'
import WithRouter from '../../../test-utils/with-router'
import Ask from './ask'

const props = {
  question: '',
  hasQuestion: false,
  canSubmitPoll: false,
  poll: {},
  answers: [],
  clearPoll: jest.fn(),
  postPoll: jest.fn(),
  updateQuestion: jest.fn(),
  onAnswerChange: jest.fn(),
  onRemoveAnswer: jest.fn(),
  updateOptions: jest.fn(),
}

describe('(Route) Ask', () => {
  describe('(Action) componentDidMount', () => {
    it('should call prop clearPoll', () => {
      render(<Ask {...props} />, {
        wrapper: WithRouter
      })

      expect(props.clearPoll).toHaveBeenCalled()
    })
  })

  describe('(Render)', () => {
    describe('with question', () => {
      it('matches snapshot', () => {
        const { asFragment } = render(<Ask {...props} question="Some question" />, {
          wrapper: WithRouter
        })

        expect(asFragment()).toMatchSnapshot()
      })
    })

    describe('when hasQuestion is true', () => {
      it('matches snapshot', () => {
        const { asFragment } = render(<Ask {...props} hasQuestion />, {
          wrapper: WithRouter
        })

        expect(asFragment()).toMatchSnapshot()
      })
    })

    describe('when canSubmitPoll is true', () => {
      it('matches snapshot', () => {
        const { asFragment } = render(<Ask {...props} canSubmitPoll />, {
          wrapper: WithRouter
        })

        expect(asFragment()).toMatchSnapshot()
      })
    })

    describe('with poll', () => {
      it('matches snapshot', () => {
        const { asFragment } = render(<Ask {...props} poll={{ question: 'Question', identifier: 'kdH98eJ' }} />, {
          wrapper: WithRouter
        })

        expect(asFragment()).toMatchSnapshot()
      })
    })

    describe('with answers', () => {
      it('matches snapshot', () => {
        const { asFragment } = render(<Ask {...props} answers={['Answer 1', 'Answer 2']} />, {
          wrapper: WithRouter
        })

        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})
