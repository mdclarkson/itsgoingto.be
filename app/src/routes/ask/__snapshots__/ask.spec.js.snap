// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) Ask (Render) when canSubmitPoll is true matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container header-container hideable"
    >
      <div
        class="header center-text"
      >
        <h1>
          <span
            class="word-rotate"
          >
            <span
              class="word-rotate_word"
            >
              What
            </span>
            <span
              class="word-rotate_word word-rotate_word--previous"
            >
              Who
            </span>
          </span>
          is it going to be?
        </h1>
      </div>
    </div>
    <div
      class="container question-container"
    >
      <div
        class="input input-question"
      >
        <label
          class="input-label input-label-question"
          for="question"
        >
          Ask a question
          <textarea
            class="input-field input-field-question js-auto-size"
            id="question"
            name="question"
            placeholder=""
            rows="1"
            style="overflow-x: hidden; word-wrap: break-word; overflow-y: hidden;"
          />
        </label>
      </div>
      <div
        class="answers hideable gone"
      >
        <label
          class="input-label input-label-answers"
        >
          Add some responses
        </label>
        <div
          class="input input-answer input-disabled"
        >
          <label
            class="input-label input-label-answer"
            for="answer-0"
          >
            1
          </label>
          <input
            class="input-field input-field-answer"
            disabled=""
            id="answer-0"
            name="answer-0"
            type="text"
          />
        </div>
      </div>
      <div>
        <div
          class="options hideable gone"
        >
          <div
            class="input-option input-option-multipleChoice"
          >
            <label
              class="input-label input-label-options input-label-multipleChoice"
              for="multiple-choice"
            >
              <input
                class="input-checkbox input-checkbox-multipleChoice"
                id="multiple-choice"
                name="multiple"
                type="checkbox"
              />
              Multiple choice responses
            </label>
          </div>
          <div
            class="input-option input-option-passphrase"
          >
            <label
              class="input-label input-label-passphrase"
              for="passphrase"
            >
              Set a passphrase (optional)
              <input
                class="input-field input-field-passphrase"
                id="passphrase"
                name="passphrase"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="input-option input-option-endPoll"
          >
            <div
              class="input-option-radios"
            >
              <input
                checked=""
                class="input-radio input-checkbox-endType"
                id="end-never"
                name="endType"
                type="checkbox"
                value="endNever"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-never"
              >
                Don't End Poll
              </label>
              <input
                class="input-radio input-checkbox-endType"
                id="end-at"
                name="endType"
                type="checkbox"
                value="endAt"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-at"
              >
                End Poll At
              </label>
              <input
                class="input-radio input-checkbox-endType"
                id="end-in"
                name="endType"
                type="radio"
                value="endIn"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-in"
              >
                End Poll In
              </label>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="actions hideable gone"
        >
          <button
            class="btn pull-right"
            data-testid="button-Create-Poll"
            type="button"
          >
            <span>
              Create Poll
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Route) Ask (Render) when hasQuestion is true matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container header-container hideable gone"
    >
      <div
        class="header center-text"
      >
        <h1>
          <span
            class="word-rotate"
          >
            <span
              class="word-rotate_word"
            >
              What
            </span>
            <span
              class="word-rotate_word word-rotate_word--previous"
            >
              Who
            </span>
          </span>
          is it going to be?
        </h1>
      </div>
    </div>
    <div
      class="container question-container move-up"
    >
      <div
        class="input input-question"
      >
        <label
          class="input-label input-label-question"
          for="question"
        >
          Ask a question
          <textarea
            class="input-field input-field-question js-auto-size"
            id="question"
            name="question"
            placeholder=""
            rows="1"
            style="overflow-x: hidden; word-wrap: break-word; overflow-y: hidden;"
          />
        </label>
      </div>
      <div
        class="answers hideable"
      >
        <label
          class="input-label input-label-answers"
        >
          Add some responses
        </label>
        <div
          class="input input-answer input-disabled"
        >
          <label
            class="input-label input-label-answer"
            for="answer-0"
          >
            1
          </label>
          <input
            class="input-field input-field-answer"
            disabled=""
            id="answer-0"
            name="answer-0"
            type="text"
          />
        </div>
      </div>
      <div>
        <div
          class="options hideable"
        >
          <div
            class="input-option input-option-multipleChoice"
          >
            <label
              class="input-label input-label-options input-label-multipleChoice"
              for="multiple-choice"
            >
              <input
                class="input-checkbox input-checkbox-multipleChoice"
                id="multiple-choice"
                name="multiple"
                type="checkbox"
              />
              Multiple choice responses
            </label>
          </div>
          <div
            class="input-option input-option-passphrase"
          >
            <label
              class="input-label input-label-passphrase"
              for="passphrase"
            >
              Set a passphrase (optional)
              <input
                class="input-field input-field-passphrase"
                id="passphrase"
                name="passphrase"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="input-option input-option-endPoll"
          >
            <div
              class="input-option-radios"
            >
              <input
                checked=""
                class="input-radio input-checkbox-endType"
                id="end-never"
                name="endType"
                type="checkbox"
                value="endNever"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-never"
              >
                Don't End Poll
              </label>
              <input
                class="input-radio input-checkbox-endType"
                id="end-at"
                name="endType"
                type="checkbox"
                value="endAt"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-at"
              >
                End Poll At
              </label>
              <input
                class="input-radio input-checkbox-endType"
                id="end-in"
                name="endType"
                type="radio"
                value="endIn"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-in"
              >
                End Poll In
              </label>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="actions hideable"
        >
          <button
            class="btn pull-right disabled"
            data-testid="button-Create-Poll"
            disabled=""
            type="button"
          >
            <span>
              Create Poll
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Route) Ask (Render) with answers matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container header-container hideable"
    >
      <div
        class="header center-text"
      >
        <h1>
          <span
            class="word-rotate"
          >
            <span
              class="word-rotate_word"
            >
              What
            </span>
            <span
              class="word-rotate_word word-rotate_word--previous"
            >
              Who
            </span>
          </span>
          is it going to be?
        </h1>
      </div>
    </div>
    <div
      class="container question-container"
    >
      <div
        class="input input-question"
      >
        <label
          class="input-label input-label-question"
          for="question"
        >
          Ask a question
          <textarea
            class="input-field input-field-question js-auto-size"
            id="question"
            name="question"
            placeholder=""
            rows="1"
            style="overflow-x: hidden; word-wrap: break-word; overflow-y: hidden;"
          />
        </label>
      </div>
      <div
        class="answers hideable gone"
      >
        <label
          class="input-label input-label-answers"
        >
          Add some responses
        </label>
        <div
          class="input input-answer"
        >
          <label
            class="input-label input-label-answer"
            for="answer-0"
          >
            1
          </label>
          <input
            class="input-field input-field-answer"
            id="answer-0"
            name="answer-0"
            type="text"
            value="Answer 1"
          />
        </div>
        <div
          class="input input-answer"
        >
          <label
            class="input-label input-label-answer"
            for="answer-1"
          >
            2
          </label>
          <input
            class="input-field input-field-answer"
            id="answer-1"
            name="answer-1"
            type="text"
            value="Answer 2"
          />
        </div>
        <div
          class="input input-answer input-disabled"
        >
          <label
            class="input-label input-label-answer"
            for="answer-2"
          >
            3
          </label>
          <input
            class="input-field input-field-answer"
            disabled=""
            id="answer-2"
            name="answer-2"
            type="text"
          />
        </div>
      </div>
      <div>
        <div
          class="options hideable gone"
        >
          <div
            class="input-option input-option-multipleChoice"
          >
            <label
              class="input-label input-label-options input-label-multipleChoice"
              for="multiple-choice"
            >
              <input
                class="input-checkbox input-checkbox-multipleChoice"
                id="multiple-choice"
                name="multiple"
                type="checkbox"
              />
              Multiple choice responses
            </label>
          </div>
          <div
            class="input-option input-option-passphrase"
          >
            <label
              class="input-label input-label-passphrase"
              for="passphrase"
            >
              Set a passphrase (optional)
              <input
                class="input-field input-field-passphrase"
                id="passphrase"
                name="passphrase"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="input-option input-option-endPoll"
          >
            <div
              class="input-option-radios"
            >
              <input
                checked=""
                class="input-radio input-checkbox-endType"
                id="end-never"
                name="endType"
                type="checkbox"
                value="endNever"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-never"
              >
                Don't End Poll
              </label>
              <input
                class="input-radio input-checkbox-endType"
                id="end-at"
                name="endType"
                type="checkbox"
                value="endAt"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-at"
              >
                End Poll At
              </label>
              <input
                class="input-radio input-checkbox-endType"
                id="end-in"
                name="endType"
                type="radio"
                value="endIn"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-in"
              >
                End Poll In
              </label>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="actions hideable gone"
        >
          <button
            class="btn pull-right disabled"
            data-testid="button-Create-Poll"
            disabled=""
            type="button"
          >
            <span>
              Create Poll
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Route) Ask (Render) with poll matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container header-container hideable"
    >
      <div
        class="header center-text"
      >
        <h1>
          <span
            class="word-rotate"
          >
            <span
              class="word-rotate_word"
            >
              What
            </span>
            <span
              class="word-rotate_word word-rotate_word--previous"
            >
              Who
            </span>
          </span>
          is it going to be?
        </h1>
      </div>
    </div>
    <div
      class="container question-container"
    >
      <div
        class="input input-question"
      >
        <label
          class="input-label input-label-question"
          for="question"
        >
          Ask a question
          <textarea
            class="input-field input-field-question js-auto-size"
            id="question"
            name="question"
            placeholder=""
            rows="1"
            style="overflow-x: hidden; word-wrap: break-word; overflow-y: hidden;"
          />
        </label>
      </div>
      <div
        class="answers hideable gone"
      >
        <label
          class="input-label input-label-answers"
        >
          Add some responses
        </label>
        <div
          class="input input-answer input-disabled"
        >
          <label
            class="input-label input-label-answer"
            for="answer-0"
          >
            1
          </label>
          <input
            class="input-field input-field-answer"
            disabled=""
            id="answer-0"
            name="answer-0"
            type="text"
          />
        </div>
      </div>
      <div>
        <div
          class="options hideable gone"
        >
          <div
            class="input-option input-option-multipleChoice"
          >
            <label
              class="input-label input-label-options input-label-multipleChoice"
              for="multiple-choice"
            >
              <input
                class="input-checkbox input-checkbox-multipleChoice"
                id="multiple-choice"
                name="multiple"
                type="checkbox"
              />
              Multiple choice responses
            </label>
          </div>
          <div
            class="input-option input-option-passphrase"
          >
            <label
              class="input-label input-label-passphrase"
              for="passphrase"
            >
              Set a passphrase (optional)
              <input
                class="input-field input-field-passphrase"
                id="passphrase"
                name="passphrase"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="input-option input-option-endPoll"
          >
            <div
              class="input-option-radios"
            >
              <input
                checked=""
                class="input-radio input-checkbox-endType"
                id="end-never"
                name="endType"
                type="checkbox"
                value="endNever"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-never"
              >
                Don't End Poll
              </label>
              <input
                class="input-radio input-checkbox-endType"
                id="end-at"
                name="endType"
                type="checkbox"
                value="endAt"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-at"
              >
                End Poll At
              </label>
              <input
                class="input-radio input-checkbox-endType"
                id="end-in"
                name="endType"
                type="radio"
                value="endIn"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-in"
              >
                End Poll In
              </label>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="actions hideable gone"
        >
          <button
            class="btn pull-right disabled"
            data-testid="button-Create-Poll"
            disabled=""
            type="button"
          >
            <span>
              Create Poll
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Route) Ask (Render) with question matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container header-container hideable"
    >
      <div
        class="header center-text"
      >
        <h1>
          <span
            class="word-rotate"
          >
            <span
              class="word-rotate_word"
            >
              What
            </span>
            <span
              class="word-rotate_word word-rotate_word--previous"
            >
              Who
            </span>
          </span>
          is it going to be?
        </h1>
      </div>
    </div>
    <div
      class="container question-container"
    >
      <div
        class="input input-question"
      >
        <label
          class="input-label input-label-question"
          for="question"
        >
          Ask a question
          <textarea
            class="input-field input-field-question js-auto-size"
            id="question"
            name="question"
            placeholder=""
            rows="1"
            style="overflow-x: hidden; word-wrap: break-word; overflow-y: hidden;"
          >
            Some question
          </textarea>
        </label>
      </div>
      <div
        class="answers hideable gone"
      >
        <label
          class="input-label input-label-answers"
        >
          Add some responses
        </label>
        <div
          class="input input-answer input-disabled"
        >
          <label
            class="input-label input-label-answer"
            for="answer-0"
          >
            1
          </label>
          <input
            class="input-field input-field-answer"
            disabled=""
            id="answer-0"
            name="answer-0"
            type="text"
          />
        </div>
      </div>
      <div>
        <div
          class="options hideable gone"
        >
          <div
            class="input-option input-option-multipleChoice"
          >
            <label
              class="input-label input-label-options input-label-multipleChoice"
              for="multiple-choice"
            >
              <input
                class="input-checkbox input-checkbox-multipleChoice"
                id="multiple-choice"
                name="multiple"
                type="checkbox"
              />
              Multiple choice responses
            </label>
          </div>
          <div
            class="input-option input-option-passphrase"
          >
            <label
              class="input-label input-label-passphrase"
              for="passphrase"
            >
              Set a passphrase (optional)
              <input
                class="input-field input-field-passphrase"
                id="passphrase"
                name="passphrase"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="input-option input-option-endPoll"
          >
            <div
              class="input-option-radios"
            >
              <input
                checked=""
                class="input-radio input-checkbox-endType"
                id="end-never"
                name="endType"
                type="checkbox"
                value="endNever"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-never"
              >
                Don't End Poll
              </label>
              <input
                class="input-radio input-checkbox-endType"
                id="end-at"
                name="endType"
                type="checkbox"
                value="endAt"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-at"
              >
                End Poll At
              </label>
              <input
                class="input-radio input-checkbox-endType"
                id="end-in"
                name="endType"
                type="radio"
                value="endIn"
              />
              <label
                class="input-label input-label-options input-label-endType"
                for="end-in"
              >
                End Poll In
              </label>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="actions hideable gone"
        >
          <button
            class="btn pull-right disabled"
            data-testid="button-Create-Poll"
            disabled=""
            type="button"
          >
            <span>
              Create Poll
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
